spark/src/main/scala/geotrellis/spark/io/cog/COGLayer.scala

```diff
@@ -53,10 +52,11 @@ object COGLayer {
      rdd: RDD[(K, V)] with Metadata[TileLayerMetadata[K]],
      baseZoom: Int,
      compression: Compression = Deflate,
-     maxTileSize: Int = 4096,
-     minZoom: Option[Int] = None
+     minZoom: Option[Int] = None,
+     options: COGLayerWriter.Options
```

Comment:
Hm, mb it makes sense to put compression into options to? 
And to have options here with a default value.

Reply:
Yeah, I think it'd make sense to put `compression` into `Options` I only didn't because that was outside the scope of the PR.

Reply:
But if you think that it should be added in this one then I can do that

Reply:
Let's do all these `options` things in a single PR (in this one).

Reply:
Okay, that sounds good.